### PR TITLE
adding pre-commit & cleaning

### PR DESCRIPTION
**Pull request by Borda**
Add pre-commit which would help with keeping Flake8 happy
Also, I would suggest enabling this bot - https://pre-commit.ci/

### CODE REVIEW

**Comment by Borda** [comment]
@manujosephv tun the CI locally and all passes, mind approving your CI :otter:

**Comment by Borda** [comment]
@manujosephv mind checking this suggestion? :otter:

**Comment by manujosephv** [comment]
@Borda  Same as the previous commit.. I'll wait for the major update from my side to merge the PR. 

Incidentally, the whole CI/CD pipeline is something I've been struggling with and would definitely welcome help in the area...

**Comment by Borda** [comment]
> Incidentally, the whole CI/CD pipeline is something I've been struggling with and would definitely welcome help in the area...

sure I think that this one without the 14d21557fdbf1adedfb020a5c5f5a11f9fedd0a6 shall be quite safe as it updates only the pipelines... said so I would suggest merging it and enabling the bot mentioned in the description which would significantly simplify your life... :otter:

**Comment by Borda** [comment]
@manujosephv rebased, could you pls check :otter:

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: Linting
 on: [push, pull_request]
 
 jobs:
-  flake8:
+  pre-commit:
```
Comment (Borda) [inline .github/workflows/linting.yml:6]: this file is the only change of this PR, all the rest it automatically generated corrections :otter:

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+default_language_version:
```
Comment (Borda) [inline .pre-commit-config.yaml:1]: ok, also adding this config is the core contributions :)

**Pull request merged**